src/components/atoms/Button/Button.jsx

```diff
@@ -1,25 +1,34 @@
 import React from 'react';
-import { bool, string } from 'prop-types';
+import PropTypes from 'prop-types';
 import { Link } from 'components';
 import cx from 'classnames';
 
-import './Button.scss';
+import styles from './Button.scss';
 
-const Button = ({ to, href, primary, ...props }) => {
-    const styleNames = cx('root', {
-        primary,
+const Button = ({ className, href, theme, to, ...props }) => {
+    const classNames = cx(styles.root, {
+        [className]: className,
```

Comment:
`className` can be moved out of the object and next to `styles.root`. As well, personally, I dont think we should allow classes to be passed in _unless_ they are mapped to classes that live in the component's style sheet.

Reply:
> As well, personally, I dont think we should allow classes to be passed in unless they are mapped to classes that live in the component's style sheet.

What's your argument? I don't see a problem with passing custom classes to components as needed. Building layout wrappers around every ui component seems a bit too purist for my taste and leads to overly verbose markup.

Reply:
Its my opinion, but I have found it easier to deal with styles on a global level when components are only concerned with their own visual states. It opens the door for people to abuse the "theming" pattern and for collisions. Also using wrappers over passing in classes has made it a lot easier to replace components in layout in my experience. 

Whats wrong with being purist? I'd rather have a little extra markup over hunting down a style collision.

Reply:
I prefer being fairly pragmatic when approaching custom, visually complex interfaces. You can be a purist in a code sample/Medium article/codepen, but that mindset doesn't lend itself to hacking together a great experience on a relatively tight timeline. Creating pure layout wrappers for _everything_ is too rigid of an assumption in an edge-case quest to *maybe* avoid style collisions. We're already using locally-scoped styles via css modules, so as long as we're diligent about *not* styling type selectors, we should very rarely (if ever) have style collisions.